static-site/app/groups/[[...groups]]/group-logo.tsx

```diff
@@ -1,9 +1,32 @@
 "use client";
 
-import React from "react";
+import React, { useEffect, useState } from "react";
 
 import usersIcon from "../../../static/logos/fa-users-solid.svg";
 
-export default function GroupLogo(): React.ReactElement {
-  return <img alt="default group logo" height="35px" src={usersIcon.src} />;
+/**
+ * A component that asynchronously loads and displays a group's logo.
+ * Starts with a default icon and updates when the group's avatar is fetched.
+ * Stays as the default icon if the avatar cannot be fetched.
+ */
+export default function GroupLogo({ name }: {
+  /** Group name */
+  name: string
+}): React.ReactElement {
+  const [imgSrc, setImgSrc] = useState(usersIcon.src);
+  const [imgAlt, setImgAlt] = useState("default group logo");
+
+  useEffect(() => {
+    async function fetchImgSrc(): Promise<void> {
+      const response = await fetch(`/charon/getSourceInfo?prefix=/groups/${name}`);
```

Comment:
_from https://github.com/nextstrain/nextstrain.org/pull/1263#discussion_r2539311362_

This is 44 calls to the server when loading nextstrain.org/groups – not great! There should be better ways to do this such as [putting the request in a server component](https://github.com/nextstrain/nextstrain.org/pull/1263#discussion_r2536028542), but that requires extra work to convert parent components to also be server components.